### PR TITLE
fix config: failed to serialize with toml

### PR DESCRIPTION
**Pull request by edouardparis**
Sorry I did not test enough the previous PR and some cases went wrong.
The error from toml crate is quite hard to understand on first catch

---
Because all values in a TOML table must be
emitted before further tables are emitted,
it implies that sub structs fields
must be at the end of the parent struct, otherwise
toml returns the error ValueAfterTable

This commit add also some macro to noise
pub key fields and duration fields that
did not failed the serialization but were
written in multiple fields instead of
simple string.

### CODE REVIEW

File: src/config.rs [hunk 1]
```diff
@@ -522,6 +529,48 @@ mod tests {
         config_res.expect_err("Deserializing an invalid toml_str");
     }
 
+    #[test]
+    fn serialize_toml_config() {
```
Comment (darosior) [inline src/config.rs:533]: :+1: blame is shared as i acked the former PR without a test :)

**Review by darosior — COMMENTED** [review]
utACK 914cdc1f7adfc76933bb4504238f67730c8fc176

File: src/config.rs [hunk 2]
```diff
@@ -522,6 +529,48 @@ mod tests {
...
+            noise_key = '087629614d227ff2b9ed5f2ce2eb7cd527d2d18f866b24009647251fce58de38'
+
+            [stakeholder_config]
+            xpub = 'xpub6AP3nZhB34Zoan3KCL9bAdnwNHdzMbskLudpbchwTfkHwnNDXYf1769gzozjgzDNUF7iwa5nCdhE5byrcx5PDKFCUDByeuqiHa382EKhcay'
+            emergency_address = 'bc1qwqdg6squsna38e46795at95yu9atm8azzmyvckulcc7kytlcckxswvvzej'
+
+            [[stakeholder_config.watchtowers]]
+            host = '127.0.0.1:1'
+            noise_key = '46084f8a7da40ef7ffc38efa5af8a33a742b90f920885d17c533bb2a0b680cb3'
+            "#.trim_start().replace("            ", "");
```
[the remark this answers is not in the document]
Comment (edouardparis) [reply src/config.rs:568]: It's for code aestheticism, if it is not trim then `in_str` must have no indentation in order to make `assert_eq!(in_str, out_str);`  succeed.
I would look like:

```rust
        let in_str = r#"
coordinator_host = '127.0.0.1:1'
coordinator_noise_key = 'd91563973102454a7830137e92d0548bc83b4ea2799f1df04622ca1307381402'
            ...
            "#
```

[the remark this answers is not in the document]
Comment (darosior) [reply src/config.rs:568]: Oh, right

**Pull request merged**